bin/api_analysis.dart

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:args/command_runner.dart';
+import 'package:collection/collection.dart';
+import 'package:http/http.dart' as http;
+import 'package:pana/src/api_analysis/common.dart';
+import 'package:pana/src/api_analysis/kind.dart';
+import 'package:pana/src/api_analysis/lower_bound_constraint_analysis.dart';
+import 'package:pana/src/api_analysis/summary.dart';
+import 'package:path/path.dart' as path;
+import 'package:pool/pool.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:retry/retry.dart';
+
+Future<void> main(List<String> arguments) async {
+  var runner = CommandRunner(
+      'api_analysis', 'A tool for analysing the public API of a dart package.')
+    ..addCommand(SummaryCommand())
+    ..addCommand(LowerBoundConstraintAnalysisCommand())
+    ..addCommand(BatchLBCAnalysisCommand());
+  await runner.run(arguments);
+}
+
+class SummaryCommand extends Command {
+  @override
+  final name = 'summary';
+  @override
+  final description = 'Displays a summary of the public API of a package.';
+  @override
+  final usage = '''Required positional arguments:
```

Comment:
Maybe, this one isn't necessary...

Reply:
I think it automagically generates `--help`

Reply:
It looks like that getter is specific to https://github.com/dart-lang/pub :
see https://github.com/dart-lang/pub/blob/bc32a30ea5c86653e2a1899613c0a19d91b9a21c/lib/src/command.dart#L47

Reply:
I tried to replicate this with https://github.com/dart-lang/pana/pull/1121/commits/c1914f227ea5c019640c828bfcc0a57cd47b0210

Reply:
Do required positional parameters have to be enclosed in angle brackets?